### ISSUE TITLE
Drawable canvas area is smaller than it should be when magnification < 100%

### ISSUE THREAD

**Issue by HiromuHota** [HiromuHota/pentaho-kettle#99]
## Expected Behavior

The whole canvas area should be drawable.

## Actual Behavior

When magnification < 100%, the drawable canvas area becomes (width * magnification, height * magnification)

![image](https://user-images.githubusercontent.com/20668349/37231204-2be3c34e-239f-11e8-986f-091c5573f3c2.png)

## Steps to Reproduce the Problem

  1. Set zoom level < 100%.

**Comment by HiromuHota** [comment]
The root cause was that the canvas is clipped by a rectangle of ( magnification, magnification ).

First, the canvas is transformed by `gc.setTransform( translationX, translationY, 0, magnification )` at `org.pentaho.di.trans.TransPainter.buildTransformationImage()` as below
```
    shadow = false;
    gc.setTransform( translationX, translationY, 0, magnification );
    gc.setAlpha( 255 );
    drawTrans( thumb );

    gc.dispose();
```

Then, the canvas is clipped by the following JavaScript code at `bundles/org.eclipse.rap.rwt/js/rwt/widgets/GC.js`

```
    _initClipping : function( x, y, width, height ) {
      this._context.clearRect( x, y, width, height );
      this._context.beginPath();
      this._context.rect( x, y, width, height );
      this._context.clip();
    },
```

**Comment by HiromuHota** [comment]
Resolved by 1b5459160312013004299872a9244c9ba1b4d414 and HiromuHota/rap@cc13fbdb6d6d537747f7fb
